src/client.js

```diff
@@ -250,12 +250,20 @@ assign(Client.prototype, {
 
     return Object.assign(poolConfig, {
       create: () => {
-        return this.acquireRawConnection().tap((connection) => {
-          connection.__knexUid = uniqueId('__knexUid');
-
-          if (poolConfig.afterCreate) {
-            return Promise.promisify(poolConfig.afterCreate)(connection);
-          }
+        const beforePromise = poolConfig.beforeCreate ?
+          Promise.promisify(poolConfig.beforeCreate)(this.connectionSettings) :
```

Comment:
Could you move `Promise.promisify(poolConfig.beforeCreate)` outside of the call so that it is only executed once?

Reply:
Actually, since this is not being exposed to the end user, keeping it as a callback is also a viable option.